xen/arch/arm/coproc/coproc.c

```diff
@@ -199,12 +308,137 @@ bool_t coproc_is_attached_to_domain(struct domain *d, const char *path)
         return false;
 
     spin_lock(&coprocs_lock);
-    is_created = coproc->ops->vcoproc_is_created(d, coproc);
+    is_created = coproc_is_created_vcoproc(d, coproc);
     spin_unlock(&coprocs_lock);
 
     return is_created;
 }
 
+struct coproc_device *coproc_alloc(struct platform_device *pdev,
+                                   const struct coproc_ops *ops)
+{
+    struct coproc_device *coproc;
+    struct device *dev = &pdev->dev;
+    struct resource *res;
+    int num_irqs, num_mmios, i, ret;
+
```

Comment:
Should we check somewhere that coproc_alloc have been already called for this platform device? Nothing prevents platform driver to call this function twice.